src/job_cache/daemon_cache.cpp

```diff
@@ -596,6 +609,44 @@ int DaemonCache::run() {
   return 0;
 }
 
+void DaemonCache::remove_corrupt_job(int64_t job_id) {
+  // First remove this job from the database so that we don't get hung up on it anymore
+  impl->jobs.remove(job_id);
+
+  // Find this job directory so we can remove all the file
+  uint8_t group_id = job_id & 0xFF;
+  std::string job_dir = wcl::join_paths(wcl::to_hex(&group_id), std::to_string(job_id));
+
+  // Iterate over these files collecting the paths to delete
+  std::vector<std::string> to_delete;
+  auto dir_res = wcl::directory_range::open(job_dir);
+  if (!dir_res) {
+    // We can keep going even with this failure but we need to at least log it
+    wcl::log::error("cleaning corrupt job: wcl::directory_range::open(%s): %s", job_dir.c_str(),
+                    strerror(dir_res.error()))();
+    return;
+  }
+
+  // Find all the entries to remove
+  for (const auto &entry : *dir_res) {
+    if (!entry) {
+      // It isn't critical that we remove this so just log the error and move on
+      wcl::log::error("cleaning corrupt job: bad entry in %s: %s", job_dir.c_str(),
+                      strerror(entry.error()))();
+      return;
+    }
+    to_delete.emplace_back(wcl::join_paths(job_dir, entry->name));
+  }
+
+  // Unlink them all
+  for (const auto &file : to_delete) {
+    unlink_no_fail(file.c_str());
+  }
+
+  // Remove the files
+  rmdir_no_fail(job_dir.c_str());
```

Comment:
per the comments above. Do we want to have the Daemon die if unlink/rmdir fail?

Reply:
I would say we don't want it to die at all in this loop.